tests/test_pasqal.py

```diff
@@ -77,40 +82,38 @@ def __init__(
         runs=10,
         variables={"t": 100, "qubits": {"q0": 1, "q1": 2, "q2": 4, "q3": 3}},
         result={"00": 5, "11": 5},
+        status=JobStatus.DONE.name,
     ) -> None:
         self.runs = runs
         self.variables = variables
         self.result = result
         self.id = str(np.random.randint(10000))
+        self.status = status
 
 
-@pytest.fixture
-def mock_job():
-    return _MockJob()
-
-
-@pytest.fixture
-def mock_batch(mock_job, seq):
-    seq_ = copy.deepcopy(seq)
-    seq_.declare_channel("rydberg_global", "rydberg_global")
-    seq_.measure()
-
-    @dataclasses.dataclass
-    class MockBatch:
-        id = "abcd"
-        status = "DONE"
-        ordered_jobs = [
-            mock_job,
+@dataclasses.dataclass
+class MockBatch:
+    id = "abcd"
+    status: str = "DONE"
+    ordered_jobs: list[_MockJob] = dataclasses.field(
+        default_factory=lambda: [
+            _MockJob(),
             _MockJob(result={"00": 10}),
             _MockJob(result={"11": 10}),
         ]
-        sequence_builder = seq_.to_abstract_repr()
+    )
+    seq_ = copy.deepcopy(_seq)
+    seq_.declare_channel("rydberg_global", "rydberg_global")
+    seq_.measure()
+    sequence_builder = seq_.to_abstract_repr()
```

Comment:
It's weird to have this within the class, why did you need it?

Reply:
It was in the fixture before. I can move it to the top level if you prefer

Note: I'm not sure we even need a valid sequence builder, I'll remove it and see if tests still pass

Reply:
I moved it to another function, let me know if that's better for you

Reply:
Yeah, looks fine :)